plugins/global-search/src/utils/filter/useAsyncFilter.ts

```diff
@@ -46,12 +47,17 @@ export function useAsyncFilter(
         })
 
         processor.on("progress", ({ results }) => {
-            startTransition(() => {
-                const uiResults = groupResults(results)
-                setState({
-                    results: uiResults,
-                    hasResults: results.length > 0,
-                    error: null,
+            groupingAbortControllerRef.current?.abort()
+            const controller = new AbortController()
+            groupingAbortControllerRef.current = controller
+
+            void groupResults(results, controller.signal).then(results => {
+                startTransition(() => {
+                    setState({
+                        results,
+                        hasResults: results.length > 0,
+                        error: null,
+                    })
```

Comment:
Could `useAsyncFilter` not do absorb `groupResults`? That is, put everything in an `entryMap` straight away? Then it doesn't need to start from scratch on every `"progress"` 🤔